### PR TITLE
Silence in `RequestOnConnBase.Payload.add` & co

### PR DESCRIPTION
**Pull request by Geod24**
See each commit / the release notes for explanation. Unfortunately I also found a regression.
This should be the last stone to get v6.1.0 (and v7.0.0) released.

### CODE REVIEW

**Review by ibuclaw — APPROVED** [review]
I agree, it looks very dubious.  Maybe wait for a second review from @joseph-wakeling-frequenz or @ben-palmer-sociomantic though.

**Comment by Geod24** [comment]
Fixes https://github.com/sociomantic-tsunami/swarm/issues/417

File: src/swarm/neo/connection/RequestOnConnBase.d
Comment (daniel-zullo) [inline src/swarm/neo/connection/RequestOnConnBase.d:427]: it should be `elem` instead. Otherwise the input parameter in the code needs to be updated

File: src/swarm/neo/connection/RequestOnConnBase.d
Comment (daniel-zullo) [inline src/swarm/neo/connection/RequestOnConnBase.d]: Ideally it should overload the function `add`  with different input parameter as both functions share the same functionality.

**Review by daniel-zullo — COMMENTED** [review]
A couple of suggestions and LGTM

File: src/swarm/neo/connection/RequestOnConnBase.d
Comment (Geod24) [reply src/swarm/neo/connection/RequestOnConnBase.d]: I think we should deprecate / remove add in the future. It exposes too much a dangerous interface.
This way we'll just have `addCopy` / `addPointer` / `addArray` (which calls `addCopy`).
Perhaps a more descriptive name can be used, such as `addScopedData` ?

Note: Turning a function into an overload set can break code, too.

File: src/swarm/neo/connection/RequestOnConnBase.d
Comment (Geod24) [reply src/swarm/neo/connection/RequestOnConnBase.d:427]: Good catch!

File: src/swarm/neo/connection/RequestOnConnBase.d
Comment (daniel-zullo) [reply src/swarm/neo/connection/RequestOnConnBase.d]: Then it's fine to keep it as it is and no need to change the name.

**Review by daniel-zullo — APPROVED** [review]

**Pull request merged**
